src/openingHours.ts

```diff
@@ -104,6 +115,150 @@ export class OpeningHours {
     return this.isOpenOn(new Date());
   }
 
+  /**
+   * Returns next day open
+   */
+  private getNextOpeningDay(date: Date): NextOpeningDay {
+    const today = date.getDay();
+    let times;
+    let i = today;
+
+    for (let key in this.openingHours) {
+      i = i === 7 ? 0 : i + 1;
+      times = this.getTimesOfDay(i);
+      if (times.length > 0) {
+        times = this.openingHours[key];
+        break;
+      }
+    }
+    return {
+      times,
+      nextDay: SIMPLE_DAYS_NAMES[i],
+      isTomorrow: i === today + 1 ? true : false,
+    };
+  }
+
+  private openAt = (formatHours, date: Date, nextDay): string | null => {
+    if (formatHours.length > 1) {
+      const now = this.constructDateFromTime(`${date.getHours()}:${date.getMinutes()}`);
+      if (nextDay) {
+        return formatHours[0].hour;
+      }
+      for (let i = 0; i < formatHours.length - 1; i = i + 2) {
+        if (now < formatHours[i].date) {
+          console.log(i);
+          return formatHours[i].hour;
+        }
+      }
+    }
+    return null;
+  };
+
+  private openUntil = (formatHours: FormatHours[], date: Date): string => {
+    if (formatHours.length > 1) {
+      const now = this.constructDateFromTime(`${date.getHours()}:${date.getMinutes()}`);
+      for (let i = 1; i < formatHours.length; i = i + 2) {
+        if (now < formatHours[i].date || now > formatHours[formatHours.length - 1].date) {
+          return formatHours[i].hour;
+        }
+      }
+    }
+    return `Aujourd'hui`;
+  };
+
+  public checkOpeningTime = (date: Date) => {
```

Comment:
We should rename this function name.

Something like `getFullOpeningTimeStatus` and add a comment above this function that explain what this function is doing : 

```
/**
* Send more information about the current opening status, the next day that it will be opened...
* /